x-pack/platform/plugins/shared/alerting/server/rules_client/lib/denormalize_artifacts.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { SavedObjectReference } from '@kbn/core/server';
+import type { Artifact } from '../../types';
+import type { DenormalizedArtifacts } from '../types';
+
+export function denormalizeArtifacts(ruleArtifacts: Artifact | undefined): {
+  artifacts: DenormalizedArtifacts;
+  references: SavedObjectReference[];
+} {
+  const references: SavedObjectReference[] = [];
+  const artifacts: DenormalizedArtifacts = {
+    dashboards: [],
+  };
+
+  if (ruleArtifacts && ruleArtifacts.dashboards) {
+    ruleArtifacts.dashboards.forEach((dashboard, i) => {
+      const refName = `dashboard_${i}`;
+      const dashboardRef = {
+        id: dashboard.id,
+        name: refName,
+        type: 'dashboard',
+      };
+      references.push(dashboardRef);
+      if (!artifacts.dashboards) {
+        artifacts.dashboards = [];
+      }
```

Comment:
```suggestion
```

Reply:
If I remove this, I get `artifacts.dashboards` possibly undefined

<img width="587" alt="Screenshot 2025-04-15 at 10 26 17" src="https://github.com/user-attachments/assets/c00d3568-fe17-4a5c-96e2-9060508d2589" />

Reply:
@mgiota I addressed this separately in a previous commit already, see e625e734d3ee1a8975ca4d5a3af6fd4e8a8e6216.

Essentially remove the check and utilize the bang operator:

```typescript
      // `artifacts.dashboards` is always defined
      artifacts.dashboards!.push({
        refId: refName,
      });
```